### PR TITLE
Add io.nanoboyadvance.NanoBoyAdvance

### PR DESCRIPTION
**Pull request by Vulcalien**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

The upstream developer is @fleroviux. [Here](https://github.com/nba-emu/NanoBoyAdvance/issues/330)'s the discussion in which submission to FlatHub was discussed.

Non-standard permissions:
- `--filesystem=host`: the emulator stores save files in the same directory of the ROM (e.g. `/path/game.gba` and `/path/game.sav`) and since ROMs may be located anywhere the user can write to, the emulator needs to be able to write save files anywhere.
- `--device=all`: for controller support.

The `.desktop` and `.metainfo.xml` files present upstream are, in the current release, outdated. So I put the newer versions here in this repository, but they should be upstream from the next release onward.

### CODE REVIEW

File: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,168 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="desktop">
+  <id>io.nanoboyadvance.NanoBoyAdvance</id>
+  <name>NanoBoyAdvance</name>
+  <summary>A cycle-accurate Nintendo Game Boy Advance emulator</summary>
```
Comment (bbhtt) [inline io.nanoboyadvance.NanoBoyAdvance.metainfo.xml:5]:
```suggestion
  <summary>Cycle-accurate Nintendo Game Boy Advance emulator</summary>
```

File: io.nanoboyadvance.NanoBoyAdvance.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15-23.08'
+sdk: org.kde.Sdk
+command: NanoBoyAdvance
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --filesystem=host # for save files
```
Comment (hfiguiere) [inline io.nanoboyadvance.NanoBoyAdvance.yml:11]: this is very broad to just "save files"

File: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    </release>
+    <release version="1.1" date="2020-05-10" type="stable">
+      <url>https://github.com/nba-emu/NanoBoyAdvance/releases/tag/v1.1</url>
+    </release>
+    <release version="1.0" date="2020-05-10" type="stable">
+      <url>https://github.com/nba-emu/NanoBoyAdvance/releases/tag/v1.0</url>
+    </release>
+  </releases>
+  <url type="homepage">https://github.com/nba-emu/NanoBoyAdvance</url>
+  <url type="bugtracker">https://github.com/nba-emu/NanoBoyAdvance</url>
```
Comment (hfiguiere) [inline io.nanoboyadvance.NanoBoyAdvance.metainfo.xml:143]: why not a URL to the issue reporter instead?

**Comment by hfiguiere** [comment]
bot, build io.nanoboyadvance.NanoBoyAdvance

File: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml (hunk 3 above)
Comment (Vulcalien) [reply io.nanoboyadvance.NanoBoyAdvance.metainfo.xml]: Just changed it to https://github.com/nba-emu/NanoBoyAdvance/issues

File: io.nanoboyadvance.NanoBoyAdvance.desktop
[the remark this answers is not in the document]
Comment (Vulcalien) [reply io.nanoboyadvance.NanoBoyAdvance.desktop:1]: True, but the latest stable release contains older versions of them. I already sent a PR to upstream, the next release should have them.

File: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml (hunk 1 above)
Comment (Vulcalien) [reply io.nanoboyadvance.NanoBoyAdvance.metainfo.xml]: Done!

**Comment by Vulcalien** [comment]
Since the build for aarch64 failed, I added a flathub.json file that skips that architecture. I think it's best to publish the app now and try to fix that error at another time.

File: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      <li>Cycle-accurate emulation of most components, including: CPU, DMA, timers, PPU and Game Pak prefetch</li>
+      <li>Passes all AGS aging cartridge tests (NBA was the first public emulator to achieve this)</li>
+      <li>Passes most tests in the mGBA test suite</li>
+      <li>Passes ARMWrestler, gba-suite and FuzzARM CPU tests</li>
+      <li>Very high compatibility, including games that require emulation of peculiar hardware edge-cases</li>
+    </ul>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://raw.githubusercontent.com/nba-emu/NanoBoyAdvance/master/docs/screenshot.png</image>
```
[the remark this answers is not in the document]
Comment (Vulcalien) [reply io.nanoboyadvance.NanoBoyAdvance.metainfo.xml]: Fixed that

**Comment by Vulcalien** [comment]
bot, build io.nanoboyadvance.NanoBoyAdvance

File: io.nanoboyadvance.NanoBoyAdvance.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      - -D USE_SYSTEM_TOML11=ON
+      - -D USE_SYSTEM_UNARR=ON
+    sources:
+      - type: git
+        url: https://github.com/nba-emu/NanoBoyAdvance
+        tag: v1.8.0
+        commit: 752d4604ce3d1f978c8fca17b6f37e0d96703ec0
+      - type: file
+        path: io.nanoboyadvance.NanoBoyAdvance.desktop
+        sha256: 9590f87b1692a5f69467046da64d2ec31b45054daa738e576432b41f505bc308
```
Comment (hfiguiere) [inline io.nanoboyadvance.NanoBoyAdvance.yml:61]:
```suggestion
```

File: io.nanoboyadvance.NanoBoyAdvance.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      - type: file
+        path: io.nanoboyadvance.NanoBoyAdvance.metainfo.xml
+        sha256: c942851ba74b540629e8d8c9331d780730a4e2bd2ad63505cf9fc562e2115694
```
Comment (hfiguiere) [inline io.nanoboyadvance.NanoBoyAdvance.yml:64]:
```suggestion
```

**Comment by Vulcalien** [comment]
bot, build io.nanoboyadvance.NanoBoyAdvance

**Comment by bbhtt** [comment]
Clicking anything under tools menu seems to crash the app. Fedora 39/GNOME 45 wayland. I don't see anything in console.

File: io.nanoboyadvance.NanoBoyAdvance.yml (hunk 2 above)
Comment (bbhtt) [reply io.nanoboyadvance.NanoBoyAdvance.yml:11]: Seems to support filechooser portal, do you still need it?

**Comment by Vulcalien** [comment]
> Clicking anything under tools menu seems to crash the app. Fedora 39/GNOME 45 wayland

I'll have to see why that happens on that distro. The tools work fine for me on Linux Mint on Wayland. However, fullscreen doesn't. I might have missed some permissions in the manifest.
